physics/smoke_dust/module_add_emiss_burn.F90

```diff
@@ -90,32 +91,39 @@ subroutine add_emis_burn(dtstep,dz8w,rho_phy,pi,                  &
     ! Constants for the fire diurnal cycle calculation
      do j=jts,jte
        do i=its,ite
-        fire_age= time_int + (fire_end_hr(i,j))*3600.
+        fire_age= time_int + (fire_end_hr(i,j)-1._kind_phys)*3600._kind_phys  !One hour delay is due to the latency of the RAVE files
+        fire_age= MAX(0._kind_phys,fire_age)
 
           SELECT CASE ( fire_type(i,j) )   !Ag, urban fires, bare land etc.
           CASE (1)
              ! these fires will have exponentially decreasing diurnal cycle,
-             ! We assume 1hr latency in ingesting the sat. data 
-             coef_bb_dc(i,j) = 1._kind_phys/((2*pi)**0.5_kind_phys * sigma_fire_dur(1) *fire_age) * &
-                           exp(- ( log(fire_age) - avg_fire_dur(1))**2 /(2*sigma_fire_dur(1)**2 ))
+             coef_bb_dc(i,j) = 1._kind_phys/((2._kind_phys*pi)**0.5_kind_phys * sigma_fire_dur(1) *fire_age) * &
```

Comment:
Recalculating this constant [the leading 1/sqrt(2pi)] for millions of grid points and timesteps is not optimal. Consider defining a constant to save a bit of compute time.

Reply:
@grantfirl A constant as below will be used. Thanks.
     coef_con = 1._kind_phys/((2._kind_phys*pi)**0.5_kind_phys * sigma_fire_dur(1) *fire_age) * &
                exp(- ( log(fire_age) - avg_fire_dur(1))**2 /(2._kind_phys*sigma_fire_dur(1)**2 ))
     do j=jts,jte
       do i=its,ite
        fire_age= time_int + (fire_end_hr(i,j)-1._kind_phys)*3600._kind_phys  !One hour delay is due to the latency of the RAVE files
        fire_age= MAX(0._kind_phys,fire_age)

          SELECT CASE ( fire_type(i,j) )   !Ag, urban fires, bare land etc.
          CASE (1)
             ! these fires will have exponentially decreasing diurnal cycle,
             coef_bb_dc(i,j) = coef_con

Reply:
@grantfirl Yes, fire_age is calculated within the loop. We should move it back into the loop.